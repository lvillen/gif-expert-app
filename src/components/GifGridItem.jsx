import React from 'react';
import PropTypes from 'prop-types';
import 'animate.css';
import { AddCategory } from './AddCategory';

export const GifGridItem = ({ title, url }) => {
  return (
    <>
      <div className="card animate__animated animate__fadeInUp">
        <img src={ url } alt={ title } />
        <p> { title } </p>
      </div>
    </>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}