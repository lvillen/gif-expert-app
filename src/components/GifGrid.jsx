import { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categoryProp }) => {
  const [images, setImages] = useState([]);

  useEffect( () => {
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?q=Lord+of+the+Rings&limit=10&api_key=wOYIa1qIXGN2erRRDRcJ8KWrqPA6XiyI';
    const response = await fetch(url);
    const { data } = await response.json();

    const gifs = data.map( img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      };
    });

    setImages( gifs );
  }

  return (
    <>
      {console.log(images)}
      <h3>{ categoryProp }</h3>
      {
        images.map( img => (
          <GifGridItem 
            key={img.id} 
            { ...img } 
          />
        ))
      }
    </>
  );
};
