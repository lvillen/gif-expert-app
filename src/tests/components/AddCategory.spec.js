import React from 'react';
import { AddCategory } from '../../components/AddCategory';
import { shallow } from 'enzyme';


describe('Tests AddCategory Component', () => {
  const setCategories = jest.fn();
  let wrapper; 

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow( <AddCategory setCategoriesProp={ setCategories }/>)

  })

  it('should match the snapshot', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  it('should change textbox', () => {
    const input = wrapper.find('input');
    const value = 'Hello world';

    input.simulate('change', { target: { value } });
  });

  it('should not call setCategories when input does not match conditions', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect( setCategories ).not.toHaveBeenCalled();
  });

  it('should call setCategories and clear input', () => {
    const value = 'Hello World';

    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect( setCategories ).toHaveBeenCalled();
    expect( wrapper.find('input').prop('value') ).toBe('');
  });
});