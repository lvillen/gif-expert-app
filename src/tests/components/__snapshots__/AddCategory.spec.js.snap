// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests AddCategory Component should match the snapshot 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
